source/common/config/grpc_mux_subscription_impl.cc

```diff
@@ -62,12 +66,14 @@ void GrpcMuxSubscriptionImpl::onConfigUpdate(
 }
 
 void GrpcMuxSubscriptionImpl::onConfigUpdateFailed(const EnvoyException* e) {
-  disableInitFetchTimeoutTimer();
   // TODO(htuch): Less fragile signal that this is failure vs. reject.
   if (e == nullptr) {
     stats_.update_failure_.inc();
     ENVOY_LOG(debug, "gRPC update for {} failed", type_url_);
   } else {
+    // fetch timeout should be disabled only when the actual timeout happens - not on network
```

Comment:
This seems very fragile to me that we are using exceptions in this way to figure out a timeout vs. not, etc. Can you rework this code to make the control flow a lot more obvious? It's possible that you might need to rework how the update failed functions work.

Reply:
Agree this is fragile. I did not try to change that because it triggers more  changes because of how the update failed functions work today and this existed earlier as well. But it would be good to clean this up. Let me try.